### ISSUE TITLE
[QUERY] Event Hub producer client cannot CreateBatch when using a multi-tenant certificate - InvalidIssuer

### ISSUE THREAD

**Issue by mikaila-msft** [Azure/azure-sdk-for-net#25306]
**Query/Question**
I have a certificate to my team's first party application. I'm creating an EventHubProducerClient and am using ClientCertificateCredential as the TokenCredential argument. I've ensured that we are sending the certificate chain by setting that ClientCertificateCredentialOption to true. The certificate we're using is one for a first party app we have for our resource provider.

When I try to call producer.CreateBatchAsync(), I get the following error.

System.UnauthorizedAccessException was thrown: Put token failed. status-code: 401, status-description: InvalidIssuer: Token issuer is invalid.

This only happens in my team's Prod environment, PPE works fine.
The issuer is fine, as this cert if used elsewhere in our project. Does anyone have insight on what could cause this error?

There is no documentation for exceptions related to Put token failures that CreateBatchAsync() can throw.

I'm not sure if this is a bug, so I just made it a query. 
Does EventHub have the ability to to handle tokens issued for multiple tenants?

**Environment:**
 - Name and version of the Library package used
 Azure.Messaging.EventHubs 5.6.2

 - Hosting platform or OS and .NET runtime version
 .NET SDK (reflecting any global.json):
 Version:   5.0.403

Runtime Environment:
 OS Version:  10.0.22000
 OS Platform: Windows

Host (useful for support):
  Version: 5.0.12

.NET runtimes installed:
  Microsoft.AspNetCore.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

 - IDE and version : [e.g. Visual Studio 16.3]
Visual Studio Enterprise 2019
Version 16.11.5

**Comment by jsquire** [comment]
Hi @mikaila-msft.  Thank you for reaching out and we regret that you're experiencing difficulties.  The Event Hubs client library doesn't participate in authorization, it only passes the supplied credential to the Event Hubs service.   The message that you're seeing would appear to indicate that the service received the credential and rejected it.

We'll need to ask the Event Hubs service team to assist; I've tagged and routed this for service attention.


//fyi:  @sjkwak,  @serkantkaraca

**Comment by mikaila-msft** [comment]
Ah, I see. Thank you Jeese.

**Comment by jsquire** [comment]
@mikaila-msft:  I was thinking about this over the weekend and something about the error message doesn't feel like Event Hubs to me.   Would you be so kind as to share the stack trace so that we can validate this isn't an error being triggered within the `Azure.Identity` library as it attempts to form the token within the credential?

**Comment by mikaila-msft** [comment]
Hey @jsquire . I was able to get in contact with @serkantkaraca and learned that it was an error from EH service. Ultimate, the reason was due to the tenant ID being passed to ClientCertificateCredential. We were passing the tenant ID of ourselves, rather than the tenant ID of the customer we want to write events to. Serkant let me know that we should be requesting tokens on behalf of the customer who has our first party app registered in their tenant.
